diff --git a/c/keys.el b/c/keys.el
index 6e70122..bc5db3a 100644
--- a/c/keys.el
+++ b/c/keys.el
@@ -16,22 +16,23 @@
    "C-S-r"
    "C-S-d"
    "C-S-i"
-   "C-q"
    "C-s"
    "C-c C-x"
    ;; "C-c C-u"
    "C-x C-e"
    "C-x C-d"
    ;; "C-x C-p"
+
    ;; "M-q"
+   "C-q"
+
+   "C-x C-q" "C-c M-y" "C-γ" ;; former
    "M-j"
    "M-t"
    "M-k"
    "M-," "M-;"
    "s-&" "s-'" "s-+" "s-," "s--" "s-0" "s-:" "s-=" "s-?" "s-C" "s-D" "s-E" "s-F" "s-H" "s-L" "s-M" "s-S" "s-^" "s-`" "s-a" "s-c" "s-d" "s-e" "s-f" "s-g" "s-h" "s-j" "s-k" "s-l" "s-m" "s-n" "s-o" "s-p" "s-q" "s-s" "s-t" "s-u" "s-v" "s-w" "s-x" "s-y" "s-z" "s-|" "s-~" "M-r"
    )
- )
-
 
 (Ox33b4O/$/set-key (meta-comma ",") #'(lambda () (interactive) (find-file "~/.emacs.d/t/k.el")))
 (Ox33b4O/$/set-key (meta-comma "f") #'(lambda () (interactive) (find-file "~/.emacs.d/t/f.el")))
@@ -83,7 +84,7 @@
 (Ox33b4O/$/set-key '("C-M-v" "M-V") 'clipboard-yank)
 (Ox33b4O/$/set-key '("C-M-c" "M-c") 'clipboard-kill-ring-save)
 (Ox33b4O/$/set-key '("C-M-x" "M-X" ) 'clipboard-kill-region)
-(Ox33b4O/$/set-key '("C-x C-q" "C-c M-y" "C-γ") 'keyboard-quit)
+;; (Ox33b4O/$/set-key '("C-x C-q" "C-c M-y" "C-γ") 'keyboard-quit)
 (Ox33b4O/$/set-key '("C-x C-d C-h") 'info)
 ;; (Ox33b4O/$/set-key "C-c C-u C-d C-a" #'(lambda () (interactive) (when (read-only-mode -8) (message "%s unlocked" (buffer-name)))))
 
